lib/site.taxonomy.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Module dependencies.
+ */
+import SiteTaxonomyTerm from './site.taxonomy.term';
+
+/**
+ * SiteTaxonomy class
+ */
+class SiteTaxonomy {
+	/**
+	 * Create a SiteTaxonomy instance
+	 *
+	 * @param {String} taxonomy - taxonomy type
+	 * @param {String} siteId - site id
+	 * @param {WPCOM} wpcom - wpcom instance
+	 * @return {Null} null
+	 */
+	constructor( taxonomy, siteId, wpcom ) {
+		if ( ! siteId ) {
+			throw new Error( '`siteId` is not correctly defined' );
```

Comment:
There's also `TypeError`, which inherits from `Error` and aligns more specifically with the semantic intent here.

>A `TypeError` is thrown when an operand or argument passed to a function is incompatible with the type expected by that operator or function.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError